phoenicis-javafx/src/main/java/org/phoenicis/javafx/views/mainwindow/apps/ApplicationFilter.java

```diff
@@ -187,6 +264,14 @@ public boolean filter(ScriptDTO script) {
             result &= CollectionUtils.isEmpty(script.getTestingOperatingSystems());
         }
 
+        /*
+         * If "Show all Operating Systems" is not selected, show only applications that fit to the used operating system
+         */
+        if (!containAllOSCompatibleApplications.getValue()) {
+            result &= script.getCompatibleOperatingSystems()
```

Comment:
I guess this will crash as well if `getCompatibleOperatingSystems()` returns `null`.